js/ScoringSet.js

```diff
@@ -397,13 +400,54 @@ export default class ScoringSet extends Backbone.Controller {
     return (this.isPassed === false);
   }
 
+  /**
+   * Define the objective for reporting purposes
+   * @protected
+   */
+  _initializeObjective() {
+    if (this.subsetParent) return;
+    const id = this.id;
+    const description = this.title;
+    const completionStatus = COMPLETION_STATE.NOTATTEMPTED.asLowerCase;
+    OfflineStorage.set('objectiveDescription', id, description);
+    if (this.isComplete) return;
+    OfflineStorage.set('objectiveStatus', id, completionStatus);
+  }
+
+  /**
+   * Reset the objective data
+   * @protected
+   */
+  _resetObjective() {
+    if (this.subsetParent || this.isComplete) return;
+    const id = this.id;
+    const completionStatus = COMPLETION_STATE.INCOMPLETE.asLowerCase;
+    OfflineStorage.set('objectiveScore', id, this.score, this.minScore, this.maxScore);
+    OfflineStorage.set('objectiveStatus', id, completionStatus);
+  }
+
+  /**
+   * Complete the objective
+   * @todo Always updates to latest data - is this desired?
+   * @protected
+   */
+  _completeObjective() {
+    if (this.subsetParent) return;
+    const id = this.id;
```

Comment:
Do we really need to set this variable?

Reply:
Force of habit to assign this way when being called more than once, but in this case the `id` getter isn't doing any calculations, so likely no performance benefit.